src/hotspot/cpu/x86/stubGenerator_x86_64_arraycopy.cpp

```diff
@@ -577,6 +579,10 @@ address StubGenerator::generate_disjoint_copy_avx3_masked(address* entry, const
     // PRE-MAIN-POST loop for aligned copy.
     __ BIND(L_entry);
 
+    __ movq(temp2, temp1);
+    __ shlq(temp2, shift);
+    __ cmpq(temp2, large_threshold);
+    __ jcc(Assembler::greaterEqual, L_copy_large);
```

Comment:
I suspect additional checks for 2.5MB array size may hit the performance of other general sizes.

Reply:
Comparing several runs of the  XorTest.copy SMALL (100K) benchmark, baseline against PR, I see an average slowdown of 1.7% (7.566 ms / op vs 7.696 ms/op)